### ISSUE TITLE
✍️ Contribution period: Ananya Srivastava

### ISSUE THREAD

**Issue by ana42742** [ersilia-os/ersilia#612]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [ ] Install the Ersilia Model Hub and test the simplest model
- [ ] Write a motivation statement to work at Ersilia
- [ ] Submit your first contribution to the Outreachy site

### Week 2 - Install and run an ML model

- [ ] Select a model from the suggested list
- [ ] Install the model in your system
- [ ] Run predictions for the EML
- [ ] Compare results with the Ersilia Model Hub implementation!

### Week 3 - Propose new models

- [ ] Suggest a new model and document it (1)
- [ ] Suggest a new model and document it (2)
- [ ] Suggest a new model and document it (3)

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by GemmaTuron** [comment]
Hello @ana42742 !

Welcome to Ersilia!
Please report here which system are you operating and whether you found any problems in the installation and run of models

Thanks

**Comment by ana42742** [comment]
Hey @GemmaTuron ! Glad to be here :) I am using a macOS Ventura with the Apple chip. I did run into some difficulty during installation while using conda, so instead I used pip. I was able to run the sample model successfully.

**Comment by ana42742** [comment]
### Ananya's Motivation Letter for Contributing to Ersilia 
I've always thought that contributing to open source software was a niche activity. After contributing to the codebase of many organisation, I have learnt that anyone can dabble in it . As a result, I applied for Outreachy, something I never would have thought to do even two months ago. I contacted a few of the former Outreachy interns after reading their blogs. I was looking to improve not only my software abilities (which I am accomplishing during the contribution period), but also my communication and teamwork skills.

I specifically chose Ersilia for three reasons. First of all, I have experience with data science and machine learning, and I have previously compared deep learning models for X-ray classification. Second, one of the former interns I spoke to had worked at Ersilia, and they advised me to apply right away. Their time at Ersilia has been extremely wonderful, and they remarked on how well-organized the tasks given to interns are. Thirdly, the atmosphere in this community is one of warmth and support.

I'm a sophomore majoring in computer science engineering with a focus in artificial intelligence. I've already taken courses in artificial intelligence, introduction to data science, and python programming, among other things. I developed a tool to automate data visualisation while working as a **research intern** at IIT-Delhi. In my research paper titled "_Classification of Chest Radiography Scans for COVID-19_”, which will soon be published in **IEEE Explore**, I compared the performance of a _VGG-16 neural network model_ and a standard CNN architecture that employed data augmentation to classify chest X-ray scans. I conducted a literature review for more than 50 studies that addressed the use of deep learning models to speed up disease detection and diagnosis. The languages and frameworks that I am proficient in are: _tensorflow, keras, numpy, opencv, NLP (text-to-speech), pandas, data visualisation (matplotlib, seaborn, plotly), Python, C++, Java, JavaScript, HTML, CSS_, amongst others.

I think that contributing to Ersilia would provide me the finest chance to enhance my machine learning and research skills. As an intern, I'm going to leverage my research and computer science expertise to the project's advantage in order to complete it. Because I have such a strong enthusiasm for this subject, I also intend to pursue a master's degree, during which time I hope to continue contributing to Ersilia.

**Comment by ana42742** [comment]
I have successfully submitted my first contribution on the Outreachy website!

**Comment by ana42742** [comment]
Hey @GemmaTuron ! Seems like the model's dependencies have deprecated long ago. I have had to install python 3.7 just to get all of its dependencies to be compatible, but even so numpy 1.19.2 is not building on my system. I installed python 3.7 using `brew install python@3.7` through x86_64. Then I pip installed `numpy 1.19.2`. Even so, upon fetching the model, it gave me numpy build error: 
```
Collecting numpy==1.19.2
  Using cached numpy-1.19.2.zip (7.3 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
ld: warning: ignoring file _configtest.o, building for macOS-x86_64 but attempting to link with file built for unknown-arm64
      Undefined symbols for architecture x86_64:
```

**Comment by ana42742** [comment]
@GemmaTuron @DhanshreeA A couple of findings, it appears that M1 ARM chips do not support python 3.7 which is what is required by the eos5qfo model. So instead of the CLI, I'm going to shift to Google Colab. Up until now I have tried solving the issue by using [pyenv](https://stackoverflow.com/questions/51551557/pyenv-build-failed-installing-python-on-macos) to manage multiple python versions, but it appears that the `model-package-installation` for this model only uses `python -m pip install numpy==1.19.2`

**Comment by ana42742** [comment]
I used the `verbose` mode to fetch the model which told me that numpy 1.19.2 wheels could not be built. So I tried using `python -m pip install numpy==1.19.2` manually (as suggested in the error) which gave me the same error.

**Comment by ana42742** [comment]
Surprisingly, `conda install numpy=1.19.2` does work! But as `ersilia fetch eos5qfo` uses the `model-package-installation` script to install numpy which uses pip, I am unable to resolve this issue. Maybe we should change it to conda instead?

**Comment by ana42742** [comment]
As mentioned previously, I will try working on #371 through Google Colab :)

**Comment by GemmaTuron** [comment]
Hi @ana42742 

Thanks for the tests!
- Have you tried the model in Colab (which is a Linux system)
- Have you tried bumping up the numpy version / python version in the model requirements? You'd do this by cloning the repository in your local system, changing the requirements file and using the option `--repo-path` at fetch time, instead of fetching from the github repo you would fetch from your own local copy (which is modified). I am unsure if this would work but could be worth a try
